### PR TITLE
Fix import cycle

### PR DESCRIPTION
**Pull request by caesarxuchao**
Removed apimahcinery's dependency on k8s.io/api, introduced in https://github.com/kubernetes/kubernetes/pull/48497#discussion_r127312690.

Fixed hack/verify-staging-imports.sh to prevent future occurrences.

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/test/api_meta_help_test.go [hunk 1]
```diff
@@ -72,13 +72,6 @@ func TestExtractList(t *testing.T) {
 			{ObjectMeta: metav1.ObjectMeta{Name: "3"}},
 		},
 	}
-	list5 := &v1.PodList{
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/test/api_meta_help_test.go]: We also have a CarpList, don't we?

File: hack/verify-staging-imports.sh [hunk 2]
```diff
@@ -65,7 +65,7 @@ function print_forbidden_imports () {
 }
 
 RC=0
-print_forbidden_imports apimachinery || RC=1
+print_forbidden_imports apimachinery should_be_leaf || RC=1
```
Comment (sttts) [inline hack/verify-staging-imports.sh]: 👍

File: staging/src/k8s.io/apimachinery/pkg/test/api_meta_help_test.go (hunk 1 above)
Comment (caesarxuchao) [inline staging/src/k8s.io/apimachinery/pkg/test/api_meta_help_test.go]: CarpList is list4 :)

**Comment by lavalamp** [comment]
/lgtm
/approve

Thanks!!

**Pull request merged**